ocs_ci/utility/rosa.py

```diff
@@ -147,11 +166,20 @@ def appliance_mode_cluster(cluster_name, ocp_version, region):
             '  public_key: "..."'
         )
     public_key_only = remove_header_footer_from_key(public_key)
+
+    subnet_ids = config.ENV_DATA["ms_provider_subnet_ids_per_region"][region][
+        "private_subnet"
+    ]
+    if not private_link:
+        subnet_ids += f",{config.ENV_DATA['ms_provider_subnet_ids_per_region'][region]['public_subnet']}"
     cmd = (
         f"rosa create service --type {addon_name} --name {cluster_name} "
-        f"--size {size} --onboarding-validation-key {public_key_only} "
+        f"--machine-cidr {machine_cidr} --size {size} "
```

Comment:
Not really but I am managing it as per the subnets created.  The concept of a private link cluster is more related use of private subnets and not with machine-cidr. usually, we use 3 public and 3 private subnets, these clusters are accessible using kubeconfigg even in any valid network, whereas private link clusters are not. 

if we don't define --machine-cidr in command it will take the default value hence to accommodate the change of custom cidr,  I have defined it as a command parameter and the value can be default cidr or custom cidr.